app/modules/net_info.c

```diff
@@ -53,6 +53,7 @@
 static int ping_callback_ref; 
 static int ping_host_count;
 static ip_addr_t ping_host_ip;
+struct ping_option *ping_opt = NULL;
```

Comment:
This is a pretty unpleasant workaround for object lifetime issues and it implies that only one `ping` can be going at once.

Looking at the LwIP code, it appears to me that you should be able to properly manage the object lifetime by hooking both the *sent and received* callbacks in the `struct ping_option` you pass to `ping_start`.